src/context.rs

```diff
@@ -380,16 +388,21 @@ fn get_current_branch(repository: &Repository) -> Option<String> {
     shorthand.map(std::string::ToString::to_string)
 }
 
-fn get_remote_branch(repository: &Repository) -> Option<String> {
+fn get_remote_repository_info(repository: &Repository) -> Option<Remote> {
     if let Ok(head) = repository.head() {
         if let Some(local_branch_ref) = head.name() {
             let remote_ref = match repository.branch_upstream_name(local_branch_ref) {
                 Ok(remote_ref) => remote_ref.as_str()?.to_owned(),
                 Err(_) => return None,
             };
 
-            let remote = remote_ref.split('/').last().map(|r| r.to_owned())?;
-            return Some(remote);
+            let v = &mut remote_ref.rsplit('/').collect::<Vec<&str>>().into_iter();
+            let remote_branch = v.next()?.to_owned();
+            let remote_name = v.next()?.to_owned();
```

Comment:
What do you think about using [`splitn`](https://doc.rust-lang.org/std/primitive.str.html#method.splitn) here? I think this will avoid issues where the branch name contains a `/`, for example `feat/some_awesome_feature`.

```suggestion
            let v mut = remote_ref.splitn(2, "/");
            let remote_name = v.next()?.to_owned();
            let remote_branch = v.next()?.to_owned();
```

Reply:
I've published new code based on your suggestion. It required some changes but your point is totally valid.

I often use branch name containing / but I did not use one for my tests.

Good catch !